### PR TITLE
add jupyter book as the documentation backbone for the project

### PR DESCRIPTION
**Pull request by tonyfast**
this work adds:

- [x] a `docs` folder for documentation
- [x] `jupyter-book` for documentation
- [x] `nox` for automation
- [x] a github action to publish github pages

### CODE REVIEW

**Comment by tonyfast** [comment]
@trallard where should be deploy to? github pages? readthedocs?

**Comment by trallard** [comment]
Let's use GH pages for now - something I had not thought of is whether we should get us a domain 🤔

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    - uses: actions/setup-python@v2
+      with:
+        python-version: 3.9  
+    - run: python -m pip install nox
+    - run: nox
+    - uses: actions/upload-artifact@v2
+      with:
+        name: docs
+        path: _build/singlehtml/   
+    # - name: Deploy
```
Comment (trallard) [inline .github/workflows/docs.yml:23]: This can be uncommented as we will go ahead an use GH pages for now

File: docs/_config.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
+title                       : Quansight Labs Jupyter CZI Accessibility Grant  
+author                      : Quansight Labs
+copyright                   : "2021"  
+logo                        : ""
```
Comment (trallard) [inline docs/_config.yml:4]: I seem to have remembered seeing a discussion about Jupyter a11y logos.... can we do some digging and see if that landed somewhere and maybe adopt a logo?

**Pull request merged**